lib/rjson/handler.rb

```diff
@@ -45,5 +57,29 @@ def process type, rest
         rest.first
       end
     end
+
+    # Recover an invalid parse tree by dropping items that we're not certain are
+    # fully recoverable
+    def recover!
+      @truncated = true
```

Comment:
I'm a little wary that `recover!` is called on *any* error, not just the truncated ones. For the Mixpanel work we're sure that this is the only error there's going to be – because the fields *are* based on valid JSON – but I don't know that we can get away with that in this general-purpose fork.

It significantly changes the cases where this library can be used, and if we're not going to be more generic then we at least have to document that.